src/clzones.cpp

```diff
@@ -732,6 +832,188 @@ void zone_data::set_temporary_disabled( const bool enabled_arg )
+        return fd_tool_magazines_loot_zone;
+    } else if( type == zone_type_LOOT_TOOLS ) {
+        return fd_tools_loot_zone;
+    } else if( type == zone_type_LOOT_TRAPS ) {
+        return fd_traps_loot_zone;
+    } else if( type == zone_type_UNLOAD_ALL ) {
+        return fd_unload_everything_loot_zone;
+    } else if( type == zone_type_LOOT_UNSORTED ) {
+        return fd_unsorted_loot_zone;
+    } else if( type == zone_type_LOOT_VEHICLE_PARTS ) {
+        return fd_vehicle_parts_loot_zone;
+    } else if( type == zone_type_LOOT_WEAPONS ) {
+        return fd_weapons_loot_zone;
+    } else if( type == zone_type_LOOT_WOOD ) {
+        return fd_wood_loot_zone;
+    } else if( type == zone_type_MINING ) {
+        return fd_mine_terrain_zone;
+    } else if( type == zone_type_MOPPING ) {
+        return fd_mop_tile_zone;
+    } else if( type == zone_type_NO_AUTO_PICKUP ) {
+        return fd_no_auto_pickup_zone;
+    } else if( type == zone_type_NO_NPC_PICKUP ) {
+        return fd_no_npc_pickup_zone;
+    } else if( type == zone_type_NPC_NO_INVESTIGATE ) {
+        return fd_npc_ignore_sounds_zone;
+    } else if( type == zone_type_NPC_INVESTIGATE_ONLY ) {
+        return fd_npc_investigation_area_zone;
+    } else if( type == zone_type_NPC_RETREAT ) {
+        return fd_npc_retreat_zone;
+    } else if( type == zone_type_SOURCE_FIREWOOD ) {
+        return fd_firewood_source_zone;
+    } else if( type == zone_type_ZONE_START_POINT ) {
+        return fd_start_point_zone;
+    } else if( type == zone_type_VEHICLE_DECONSTRUCT ) {
+        return fd_vehicle_deconstruct_zone;
+    } else if( type == zone_type_VEHICLE_PATROL ) {
+        return fd_vehicle_patrol_zone;
+    } else if( type == zone_type_VEHICLE_REPAIR ) {
+        return fd_vehicle_repair_zone;
+    } else {
+        return fd_null;
+    }
+}
+
+void zone_data::toggle_display()
+{
+    if( this->is_vehicle ) {
+        popup( colorize( _( "Zones tied to vehicles cannot be displayed" ), c_magenta ) );
+    }
+
+    is_displayed = !is_displayed;
+
+    std::unique_ptr<tinymap> p_update_tmap = std::make_unique<tinymap>();
+    tinymap &update_tmap = *p_update_tmap;
+
+    //### failing to get project_remain to accept my parameters, and so resorting to crude alternative.
+    // Thought it should be:
+    // point_abs_sm quotient;
+    // tripoint_sm_ms remainder;
+    // std::tie(quotient, remainder) = coords::project_remain<coords::sm>(get_start_point());
+    const tripoint_abs_sm sm_start_pos = coords::project_to<coords::sm>( get_start_point() );
+    tripoint_abs_ms ms_start_projection = coords::project_to < coords::ms >( sm_start_pos );
+    tripoint_rel_ms start_remainder = get_start_point() - ms_start_projection;
+    // Assuming the code downstream can handle any relative location of start and end.
+    tripoint_rel_ms end_remainder = get_end_point() - ms_start_projection;
+
+    update_tmap.load( sm_start_pos, true );
```

Comment:
> Defining a zone covering more than one map (don't know yet if it's submap or world tile) causes only parts of it to be displayed, and crashes the game if very large.

This will only load part of the map if the zone is large so iteration through all affected submaps is needed.

Reply:
Thanks. Will work on it.